conn.go

```diff
@@ -149,11 +149,7 @@ func (cn *conn) handleDriverSettings(o values) (err error) {
 	if err != nil {
 		return err
 	}
-	err = boolSetting("binary_parameters", &cn.binaryParameters)
-	if err != nil {
-		return err
-	}
-	return nil
+	return boolSetting("binary_parameters", &cn.binaryParameters)
```

Comment:
I don't quite agree with this change.  Now the last setting is different from the rest just to save a couple of lines of code.

Reply:
Why? Isn't writing
```go
err = foo()
if err != nil {
    return err
}
return nil
```
the same as
```go
booleanVar = foo()
if booleanVar == true {
    return true
}
return false
```
? Would you agree that the latter is bad style?

Reply:
> Would you agree that the latter is bad style?

That's not quite the same thing.  What I'm saying (to try and change your analogy to what I think would be comparable) is that when you go from this:

```Go
if foo() {
    return true
}
if bar() {
    return true
}
return false
```

to this:

```Go
if foo() {
    return true
}
return bar()
```

you're making the last check "special".  If somebody wants to add one more check and it feels natural to do it at the end, they suddenly have to change the structure of the code:

```Go
if foo() {
    return true
}
if bar() {
    return true
}
return baz()
```


And in this context I think it actually makes a big difference because the code is essentially just a list of options.

Reply:
FWIW, I don't think this is a bit worth picking - do what the linter says and don't overthink it.

Reply:
@johto I thought like that before, but I've been using megacheck at work (with quite a few bugs found by it), and I have to say that I've noticed two things:
1. Code like that doesn't really change that often.
2. When it does, it isn't as painful as I had anticipated it to be.

@tamird I tend to agree. It is kinda like arguing over gofmt, just with a third-party tool.

Reply:
OK.  Consider my complaint overruled.  I can see the value in having the codebase be completely lint clean.